### PR TITLE
[ADAM-848] TwoBitFile now support nBlocks and maskBlocks

### LINKED ISSUE #848: TwoBitFile does not take into account N blocks
Quite embarrassing on my part, but driven by rushing to publish the O'Reilly book.  Will rectify this very shortly.

### PR DESCRIPTION
**Pull request by laserson**
Fixes #848.

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/BroadcastRegionJoin.scala [hunk 1]
```diff
@@ -165,7 +165,7 @@ object BroadcastRegionJoin extends RegionJoin {
  *
  * @param regions The input-set of regions.
  */
-private[rdd] class NonoverlappingRegions(regions: Iterable[ReferenceRegion]) extends Serializable {
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/BroadcastRegionJoin.scala]: Why are we making these public? IMO, they should stay package private or move to the `org.bdgenomics.adam.models` package.

File: adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala [hunk 2]
```diff
@@ -130,23 +169,23 @@ object TwoBitRecord {
...
-    })
+      ReferenceRegion(name, maskBlockStart, maskBlockStart + maskBlockSize) -> None
+    })))
     val dnaOffset = maskBlockArraysOffset + (maskBlockCount * TwoBitFile.PER_BLOCK_SIZE) + TwoBitFile.SEQ_RECORD_RESERVED_SIZE
     TwoBitRecord(dnaSize, nBlocks, maskBlocks, dnaOffset)
   }
 }
 
-case class TwoBitRecord(dnaSize: Int, nBlocks: Seq[ReferenceRegion], maskBlocks: Seq[ReferenceRegion], dnaOffset: Int)
+case class TwoBitRecord(dnaSize: Int, nBlocks: Option[NonoverlappingRegions], maskBlocks: Option[NonoverlappingRegions], dnaOffset: Int)
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala]: This can be private as well, no?

**Comment by fnothaft** [comment]
Other than nits around packaging, LGTM!

File: adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala (hunk 2 above)
Comment (laserson) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/TwoBitFile.scala]: yep, done

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/BroadcastRegionJoin.scala (hunk 1 above)
Comment (laserson) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/BroadcastRegionJoin.scala]: Yeah, couldn't decide what to do with this.  `models` doesn't exactly seem like the right place for it IMO either.  What about `org.bdgenomics.adam.util`?

Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/BroadcastRegionJoin.scala]: I mean, it is a model, right? Mostly just a container with an invariant (regions shouldn't overlap).

Comment (laserson) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/BroadcastRegionJoin.scala]: lol, true.  k, I'll move it there...

**Comment by laserson** [comment]
Ok, ready for squash and merge if you are.

**Comment by laserson** [comment]
(and rebase)

**Comment by fnothaft** [comment]
That sounds good! Can you squash and rebase and I will merge?

File: adam-core/src/main/scala/org/bdgenomics/adam/models/NonoverlappingRegions.scala [hunk 3]
```diff
@@ -0,0 +1,194 @@
+package org.bdgenomics.adam.models
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/NonoverlappingRegions.scala]: Also, add the license header. (You can do this with `./scripts/format-source`).

Comment (laserson) [inline adam-core/src/main/scala/org/bdgenomics/adam/models/NonoverlappingRegions.scala]: My favorite script :)  Haven't run that guy in a while

**Comment by laserson** [comment]
Jenkins, retest this please

**Comment by laserson** [comment]
We should be good to go on this one.

**Comment by laserson** [comment]
Jenkins, retest this please

**Comment by laserson** [comment]
Jenkins, retest this please

**Comment by laserson** [comment]
Jenkins, retest this please

**Comment by laserson** [comment]
Jenkins, retest this please

**Comment by laserson** [comment]
(it's hard to stay this polite to Jenkins)

**Comment by laserson** [comment]
Hallelujah, @fnothaft, think we're good to go.

**Comment by fnothaft** [comment]
Thank you @laserson! I've just merged.

**Pull request merged**